### PR TITLE
Week 2 andrew luna

### PR DESCRIPTION
**Pull request by lundrew**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -9,37 +9,95 @@
...
+describe("divisibleBy3", () => {
+    it("takes in a number as an argument and decides if the number is evenly divisible by 3 or not", () => {
+      const num1 = []
+      const num2 = []
+      const num3 = []
+      expect(divisibleBy3(num1))
+      expect(divisibleBy3(num2))
+      expect(divisibleBy3(num3))
+    })
+  })
```
Comment (elysemontano) [inline code-challenges.test.js:72]: The setup for this test looks great, however, this test is not comparing against and expected output which will result in a guaranteed pass.  Make sure to include a .toEqual with your expected output such as: `expect(divisibleBy3(num1)).toEqual("divisble by 3")`

File: code-challenges.test.js [hunk 2]
```diff
@@ -9,37 +9,95 @@
...
 
 // b) Create the function that makes the test pass.
+// // this is the passing test
+const divisibleBy3 = (number) => {
+    if  (number % 3 === 0){
+        return "divisble by 3"
+    } else{
+      return "not divisible by 3"
+    }
+  }
```
Comment (elysemontano) [inline code-challenges.test.js:90]: Awesome!  Logic looks great!  You can make your return more dynamic by using string interpolation.

File: code-challenges.test.js [hunk 3]
```diff
@@ -49,9 +107,35 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
-// b) Create the function that makes the test pass.
-
+describe("upperCase", () => {
+    it("takes in an array of words and returns an array with all the words capitalized.", () => {
+      const randomNouns1 = []
+      const randomNouns2 = []
+      expect(upperCase(randomNouns1))
+      expect(upperCase(randomNouns2))
+    })
+  })
```
Comment (elysemontano) [inline code-challenges.test.js:117]: Just like before, make sure to compare to an expected output.  Also, your test variables inside the test are set to an empty array.  I like the placement that you have for those test variables since they are within the scope you are working with, just be sure that you copy the test data into those variables as well.

File: code-challenges.test.js [hunk 4]
```diff
@@ -49,9 +107,35 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
...
+  // place a param called "string"
+  // return the param and use the map () method to access the values
+  // return the value's first character and use the toUpperCase() method
+
+// // b) Create the function that makes the test pass.
+const upperCase = (string) => {
+  return string.map(value=> {
+    return value.charAt(0).toUpperCase() + value.substring(1);
+  })
+}
```
Comment (elysemontano) [inline code-challenges.test.js:130]: Excellent!  I would usually recommend to have your param be named according to the data type that is being passed into your function, in this case an array is the input.

File: code-challenges.test.js [hunk 5]
```diff
@@ -65,5 +149,43 @@ const vowelTester2 = "academy"
...
+
+const vowelSpotter = (value) => {
+  let foundIndexes = []
+  for(let i = 0; i < value.length ; i++) {
+    if (vowels.indexOf(value[i]) > -1){
+      foundIndexes.push(i);
+    }
+  } 
+  return foundIndexes.sort()[0]
+}
```
Comment (elysemontano) [inline code-challenges.test.js:183]: Awesome, great job!  In this case, since you have index set to a variable already, there is no need to push it into an array and return the first index of that sorted array.  You can simply return i within your conditional.

File: coding-concepts.js [hunk 6]
```diff
@@ -8,39 +8,38 @@
 // --------------------1) What will this log?
 
 var cohort = "Bravo 2022"
-// console.log(cohort.split(""))
+console.log(cohort.split(""))
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: "B" "r" "a" "v" "o" "2" "0" "2" "2"
+// b) Verify and explain: My answer was correct because the split() method breaks the string up into a list of substrings.
```
Comment (elysemontano) [inline coding-concepts.js:14]: .split will turn a string into an array.

File: coding-concepts.js [hunk 7]
```diff
@@ -67,7 +66,9 @@ class Learn {
   }
 }
 var learnStudent = new Learn("George")
-// console.log(learnStudent)
+console.log(learnStudent)
+
+// a) Your answer: student: "George", cohort: "Bravo", year: 2022
+// b) Verify and explain: My answer was correct because we are creating a new variable based off of the "Learn" class template. With the parameter as "name" and "this.student = name" we can create a new variable or class and pass an argument to "name", which gives "name" a new value such as "George".
```
Comment (elysemontano) [inline coding-concepts.js:72]: Great work on this section!

File: interview-questions.md [hunk 8]
```diff
@@ -8,62 +8,69 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-  Your answer:
+  Your answer: The required parameters are the ones that need to be pushed from the argument. When an argument is passed, the map() function will alter the original array. 
 
-  Researched answer:
+  Researched answer: the array method called map(), calls a specified function for every array element and then creates a new array with that. This map() method does not change anything to the original array but creates a new array. We use this if we want to execute a different function to an array or an element within that array.
```
Comment (elysemontano) [inline interview-questions.md:21]: The predefined parameters in this case is referencing what params you would set when writing .map which is currentValue, index and array.  `array.map((value, index) => value`

File: interview-questions.md [hunk 9]
```diff
@@ -8,62 +8,69 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 6. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: Hoisting is related to the arrangement of code and how variables are accessed. 
 
-  Researched answer:
+  Researched answer: Hoisting is the process of moving the declaration of variables, functions or classes to the top of their scope prior to the execution of the code. It gives us the advantage that wherever the variables etc.. are declared, they are then moved to the top of their scope and does not matter if their scope is global or local.
```
Comment (elysemontano) [inline interview-questions.md:55]: Great job on this section.  Both your answers and researched answers were very well worded.

**Review by elysemontano — APPROVED** [review]
Great work Andrew!  Your code concepts and interview answers were very well explained and your logic for the code challenges looked great.  I would focus on the concepts of tests and how they work.  You have a great setup, but need to make sure that you are testing the inputs as they are passed into a function against an expected output.  Great job!
